bliss/models/encoder.py

```diff
@@ -126,18 +134,19 @@ def _sample_class_weights(class_weights, n_samples=1):
     return cat_rv.sample((n_samples,)).squeeze()
 
 
-def _get_tile_coords(slen, tile_slen):
+def _get_tile_coords(height, width, tile_slen):
     """This records (x0, x1) indices each image tile comes from."""
 
-    nptiles1 = int(slen / tile_slen)
-    n_ptiles = nptiles1 ** 2
-
-    def return_coords(i):
-        return [(i // nptiles1) * tile_slen, (i % nptiles1) * tile_slen]
+    n_ptiles1 = int(height / tile_slen)
+    n_ptiles2 = int(width / tile_slen)
 
-    tile_coords = torch.LongTensor([return_coords(i) for i in range(n_ptiles)])
-
-    return tile_coords
+    tile_coords = []
+    for i in range(n_ptiles1):
+        for j in range(n_ptiles2):
+            tile_coords.append(
+                ((i % n_ptiles1) * tile_slen, (j % n_ptiles2) * tile_slen)
+            )
+    return torch.LongTensor(tile_coords)
```

Comment:
Is this tensor on the cpu?

Reply:
Yes, but then it's moved to the GPU when returned using `type_as`

Reply:
Is there a way to create it on the GPU in the first place?

Reply:
yup I just pushed a change to do this